File: infinispan-print-configuration/src/main/java/InfinispanPrintConfiguration.java
import java.io.IOException;

import org.infinispan.Cache;
import org.infinispan.CacheImpl;
import org.infinispan.manager.DefaultCacheManager;
import org.infinispan.manager.EmbeddedCacheManager;

public class InfinispanPrintConfiguration {
    public static void main(String[] args) throws IOException {
        EmbeddedCacheManager manager =
            new DefaultCacheManager("infinispan.xml");
        Cache<?, ?> cache = manager.getCache();

        System.out.println(cache
                           .getCacheConfiguration()
                           .toString());

        System.out.println(((CacheImpl) cache).getConfigurationAsXmlString());

        System.out.println(((CacheImpl) manager.getCache("defaultNamedCache")).getConfigurationAsXmlString());

        manager.stop();
    }
}

File: infinispan-print-configuration/src/main/scala/InfinispanPrintConfiguration.scala
import org.infinispan.CacheImpl
import org.infinispan.manager.DefaultCacheManager

object InfinispanPrintConfigurationScala {
  def _main(args: Array[String]): Unit = {
    val manager = new DefaultCacheManager("infinispan.xml")
    val defaultCache = manager.getCache[Any, Any]
    val defaultNamedCache = manager.getCache[Any, Any]("defaultNamedCache")
    val namedCache = manager.getCache[Any, Any]("namedCache")

    val caches = List(defaultCache, defaultNamedCache, namedCache)

    println(defaultCache.getCacheConfiguration)
    println(defaultCache.asInstanceOf[CacheImpl[_, _]].getConfigurationAsXmlString)

    caches.foreach(_.stop())
    manager.stop()
  }
}
